pkgs/desktops/gnome-3/core/gucharmap/default.nix

```diff
@@ -12,16 +13,17 @@ stdenv.mkDerivation rec {
 
   propagatedUserEnvPkgs = [ gnome3.gnome_themes_standard ];
 
-  preConfigure = "substituteInPlace ./configure --replace /usr/bin/file ${file}/bin/file";
+  preConfigure = "patchShebangs gucharmap/gen-guch-unicode-tables.pl";
 
-  buildInputs = [ pkgconfig gtk3 intltool itstool glib appdata-tools
-                  gnome3.yelp_tools libxml2 file desktop_file_utils
-                  gnome3.gsettings_desktop_schemas makeWrapper ];
+  nativeBuildInputs = [ pkgconfig wrapGAppsHook unzip ];
 
-  preFixup = ''
-    wrapProgram "$out/bin/gucharmap" \
-      --prefix XDG_DATA_DIRS : "${gnome3.gnome_themes_standard}/share:$out/share:$GSETTINGS_SCHEMAS_PATH"
-  '';
+  buildInputs = [ gtk3 intltool itstool glib appdata-tools
+                  gnome3.yelp_tools libxml2 desktop_file_utils
+                  gnome3.gsettings_desktop_schemas ];
+
+  unicode-data = callPackage ./unicode-data.nix {};
```

Comment:
This package seems useful outside gucharmap. For example, https://github.com/NixOS/nixpkgs/blob/26033a43/pkgs/tools/misc/unicode/default.nix could use it. However, I don't mind if it remains private for now.

Reply:
Yea.
However, I'm not sure which files it should contain (different apps may require different ucd files) and should it be zipped or not.
For example, `gucharmap` requires `Unihan.zip` to be zipped, but the contents of `UCD.zip` (`Blocks.txt`, `UnicodeData.txt`, ...) should be not.

Reply:
Arch package might be used as an inspiration: https://www.archlinux.org/packages/extra/any/unicode-character-database/.